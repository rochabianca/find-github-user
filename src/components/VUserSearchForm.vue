<template>
  <form @submit.prevent="$emit('submitted', username)">
    <input data-test="search-input" v-model="username" type="text" />
    <button data-test="search-button" type="submit">Submit</button>
  </form>
</template>

<script>
export default {
  name: 'UserSearchForm',
  data() {
    return {
      username: ''
    }
  }
}
</script>

<style>
</style>
