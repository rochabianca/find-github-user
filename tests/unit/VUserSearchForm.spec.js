import { shallowMount } from '@vue/test-utils';
import VUserSearchForm from '@/components/VUserSearchForm';

describe('VUserSearchForm', () => {
  const build = () => {
    const wrapper = shallowMount(VUserSearchForm);

    return {
      wrapper,
      input: () => wrapper.find('[data-test="search-input"]'),
      button: () => wrapper.find('[data-test="search-button"')
    };
  };

  it('renders the component', () => {
    const { wrapper } = build();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders main child components', () => {
    const { input, button } = build();

    expect(input().exists()).toBe(true);
    expect(button().exists()).toBe(true);
  });

  it('calls "submitted" event when submitting form', () => {
    const expectedUser = 'kuroski';
    const { wrapper, button, input } = build();
    input().element.value = expectedUser;

    input().trigger('input');
    button().trigger('click');
    button().trigger('submit');

    expect(wrapper.emitted().submitted[0]).toEqual([expectedUser]);
  });
});
