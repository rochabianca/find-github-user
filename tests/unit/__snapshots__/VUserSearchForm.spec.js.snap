// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUserSearchForm renders the component 1`] = `<form><input data-test="search-input" type="text"> <button data-test="search-button" type="submit">Submit</button></form>`;
